src/twister2/specification_processor.py

```diff
@@ -117,7 +117,14 @@ def prepare_spec_dict(self, platform: PlatformSpecification, scenario: str) -> d
 
         test_spec_dict['name'] = self.item.name
         test_spec_dict['original_name'] = self.item.originalname  # type: ignore[attr-defined]
-        test_spec_dict['source_dir'] = self.test_directory_path
+        if 'source_dir' in test_spec_dict:
+            if os.path.isabs(test_spec_dict['source_dir']):
+                test_spec_dict['source_dir'] = Path(test_spec_dict['source_dir'])
```

Comment:
Maybe it worth to add `os.path.expanduser` to allow to use "~/" in a path.

Reply:
Changed. I also added `os.path.expandvars` to allow using path with env variable. Thanks